vhd/lib/libvhd.c

```diff
@@ -117,6 +118,24 @@ old_clear_bit(volatile char *addr, int nr)
 	((uint32_t *)addr)[nr >> 5] &= ~(1 << (nr & 31));
 }
 
+int
+open_optional_odirect(const char *pathname, int flags, ...)
+{
+	mode_t mode;
+	if (flags & O_CREAT) {
```

Comment:
I wonder if this complexity is worth it - it'd be easier to make the wrapper always take a mode_t (which could be passed as -1 in the calls which are not O_CREAT) - although if Mark is happy then I have no objections.